mm2src/coins/qrc20/qrc20_tests.rs

```diff
@@ -174,37 +174,48 @@ fn test_validate_maker_payment() {
         .validate_maker_payment(input.clone())
         .wait()
         .unwrap_err()
-        .to_string();
+        .into_inner();
     log!("error: {:?}", error);
-    assert!(
-        error.contains("Payment tx 0x9e032d4b0090a11dc40fe6c47601499a35d55fbb was sent from wrong address, expected 0x783cf0be521101942da509846ea476e683aad832")
-    );
+    assert!(matches!(error, ValidatePaymentError::WrongPaymentTx(_)));
+    if let ValidatePaymentError::WrongPaymentTx(error) = error {
+        assert!(error
+            .contains("Payment tx 0x9e032d4b0090a11dc40fe6c47601499a35d55fbb was sent from wrong address, expected 0x783cf0be521101942da509846ea476e683aad832"))
+    }
```

Comment:
This can be done in more convenient way like:

```rs
// assert!(matches!(error, ValidatePaymentError::WrongPaymentTx(_))); No longer needed
if let ValidatePaymentError::WrongPaymentTx(error) = error {
    assert!(error
        .contains("Payment tx 0x9e032d4b0090a11dc40fe6c47601499a35d55fbb was sent from wrong address, expected 0x783cf0be521101942da509846ea476e683aad832"))
} else {
    panic!("explanation");
    // or
    assert!(false, "explanation");
}
```

I see this usage exists in some other functions(in this module) as well. Could you please update them as well?

Reply:
I see you guys already talked on this point. Will mark this as resolved.